### PR TITLE
Fixes join test cases

### PR DESCRIPTION
**Pull request by johnedquinn**
# Description

Picked out the relevant ones from #122. See the description there.

# License
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-tests-data/eval/query/join/joins.ion [hunk 1]
```diff
@@ -593,7 +593,7 @@ join::[
         {
```
Comment (jpschorr) [inline partiql-tests-data/eval/query/join/joins.ion:593]: This test logically corresponds to something like (I didn't rewrite the query, just added SQL tables):

```
CREATE TABLE t2 (
    name varchar(255),
    n int
);

CREATE TABLE t3 (
    name varchar(255),
    n int
);

INSERT INTO t2 (name, n) VALUES 
	('bb', 12), ('cc', 22), ('ee', 42);
INSERT INTO t3 (name, n) VALUES 
	('bb', 13), ('cc', 23), ('dd', 33);

SELECT * 
FROM 
		(SELECT name, n as s2_n, 2 as s2_2 FROM t2) as s2 
    LEFT JOIN 
	    (SELECT name, n as s3_n, 3 as s3_2 FROM t3) s3 ON s2.name = s3.name
```


And here we assert the result is 
```
<< 
    {'name': 'bb', 'n': 12, 'name':'bb', 'b':13},
    {'name': 'cc', 'n': 22, 'name':'cc', 'b':23},
    {'name': 'ee', 'n': 42, _2:null},
>>
```


The SQL-equivalent result might be encoded as 
```
<< 
    {'name': 'bb', 'n': 12, 'name':'bb', 'b':13},
    {'name': 'cc', 'n': 22, 'name':'cc', 'b':23},
    {'name': 'ee', 'n': 42, 'name':null, 'n':null},
>>
```

Comment (johnedquinn) [reply partiql-tests-data/eval/query/join/joins.ion:593]: Exactly. Here's a corresponding [DB Fiddle](https://www.db-fiddle.com/f/dYZcyMk7tEJjABmXY3Thts/1).

File: partiql-tests-data/eval/query/join/joins.ion [hunk 2]
```diff
@@ -657,7 +657,9 @@ join::[
           name:"ee",
           s2_n:42,
           s2_2:2,
-          _4:null
```
[the remark this answers is not in the document]
Comment (johnedquinn) [reply partiql-tests-data/eval/query/join/joins.ion:660]: Exactly. Here's a corresponding [DB Fiddle](https://www.db-fiddle.com/f/dYZcyMk7tEJjABmXY3Thts/2).

File: partiql-tests-data/eval/query/select/select.ion [hunk 3]
```diff
@@ -350,7 +350,8 @@ select_join::[
       result:EvaluationSuccess,
       output:$bag::[
         {
-          id:"7"
+          id:"7",
+          title: null
         }
       ]
     }
```
[the remark this answers is not in the document]
Comment (johnedquinn) [reply partiql-tests-data/eval/query/select/select.ion:357]: Here's a corresponding [DB Fiddle](https://www.db-fiddle.com/f/dYZcyMk7tEJjABmXY3Thts/3).

File: partiql-tests-data/eval/query/select/select.ion [hunk 4]
```diff
@@ -363,7 +364,8 @@ select_join::[
       result:EvaluationSuccess,
```
[the remark this answers is not in the document]
Comment (johnedquinn) [reply partiql-tests-data/eval/query/select/select.ion:364]: Here's a corresponding [DB Fiddle](https://www.db-fiddle.com/f/dYZcyMk7tEJjABmXY3Thts/4).

**Review by jpschorr — APPROVED** [review]

**Pull request merged**